rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/nodesource/serialization/ExportToCatalogConfirmWindow.java

```diff
@@ -0,0 +1,356 @@
+                        String nodeSourceName = nodeSource.get("name").isString().stringValue();
+                        if (nodeSourceName.equals(ExportToCatalogConfirmWindow.this.nodeSourceName)) {
+                            isRevision = true;
+                            catalogRequestBuilder.sendRequestToCatalog("buckets/" + bucketName + "/resources/" +
+                                                                       nodeSourceName + "/revisions",
+                                                                       new RequestCallback() {
+                                                                           @Override
+                                                                           public void onResponseReceived(
+                                                                                   Request request, Response response) {
+                                                                               JSONArray nodeSourceRevisions = JSONParser.parseStrict(response.getText())
+                                                                                                                         .isArray();
+                                                                               JSONObject lastNodeSourceRevision = nodeSourceRevisions.get(0)
+                                                                                                                                      .isObject();
+                                                                               revisionLabel.setContents("(Last Revision: " +
+                                                                                                         lastNodeSourceRevision.get("commit_time")
+                                                                                                                               .isString()
+                                                                                                                               .stringValue() +
+                                                                                                         " \"" +
+                                                                                                         lastNodeSourceRevision.get("commit_message")
+                                                                                                                               .isString()
+                                                                                                                               .stringValue() +
+                                                                                                         "\")");
+                                                                           }
+
+                                                                           @Override
+                                                                           public void onError(Request request,
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)